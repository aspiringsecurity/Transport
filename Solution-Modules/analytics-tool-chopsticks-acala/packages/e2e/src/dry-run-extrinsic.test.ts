import { describe, expect, it } from 'vitest'

import { api, chain, dev, env, setupApi, testingPairs } from './helper'

setupApi({
  ...env.mandala,
  mockSignatureHost: true,
})

describe('dry-run-extrinsic', () => {
  it('dry run extrinsic', async () => {
    const properties = await chain.api.chainProperties
    const { alice, bob } = testingPairs('ed25519', properties.ss58Format)

    await dev.setStorage({
      System: {
        Account: [[[alice.address], { data: { free: 1000 * 1e12 } }]],
      },
    })
    const extrinsic = await api.tx.balances.transfer(bob.address, 1e12).signAsync(alice)
    const { outcome, storageDiff } = await chain.dryRunExtrinsic(extrinsic.toHex())
    expect(outcome.toHuman()).toMatchSnapshot()
    expect(storageDiff).toMatchSnapshot()
  })

  it('dry run extrinsic with fake signature', async () => {
    const ALICE = '5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu'

    await dev.setStorage({ Sudo: { Key: ALICE } })

    // sudo.sudo(system.fillBlock(10000000))
    const call = '0xff00000080969800'
    const { outcome, storageDiff } = await chain.dryRunExtrinsic({ call, address: ALICE })

    expect(outcome.toHuman()).toMatchSnapshot()
    expect(storageDiff).toMatchSnapshot()
  })
})
