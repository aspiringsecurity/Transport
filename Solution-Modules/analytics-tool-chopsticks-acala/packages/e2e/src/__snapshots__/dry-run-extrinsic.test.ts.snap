// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dry-run-extrinsic > dry run extrinsic 1`] = `
{
  "Ok": {
    "Ok": [],
  },
}
`;

exports[`dry-run-extrinsic > dry run extrinsic 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850",
    "0x0d000000",
  ],
  [
    "0x3a7472616e73616374696f6e5f6c6576656c3a",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da98a437914f3dd43c1f1f4289c5f1baa47d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69",
    "0x0000000000000000010000000000000000204aa9d10100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x3a65787472696e7369635f696e646578",
    "0x03000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9280b197190fc1a763257f65171ea366e88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
    "0x0100000000000000000000000000000007766a2d928c03000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9bf9f5fbbef06491434a79dd19744ab376d6f646c6163612f63706f740000000000000000000000000000000000000000",
    "0x00000000000000000100000000000000c3e91b2a3b0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a146c726478796bad0b9cabd7481dbe64983",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7",
    "0x34020b0a00016d6f646c6163612f63647074000000000000000000000000000000000000000016cd1b0000000000000000000000000000020b0000016d6f646c6163612f63647074000000000000000000000000000000000000000016cd1b0000000000000000000000000000020b0a00016d6f646c6163612f636470740000000000000000000000000000000000000000e737fa0100000000000000000000000000020b0a00016d6f646c6163612f6364707400000000000000000000000000000000000000004d125c100000000000000000000000000000000000000000001c6c090000000002000000010000000000000000000000000002000000020000000a0888dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eef9f9b6a20300000000000000000000000000020000000a0288dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eed17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae690010a5d4e800000000000000000000000000020000000a076d6f646c6163612f747273790000000000000000000000000000000000000000946192e80200000000000000000000000000020000001406946192e80200000000000000000000000000020000000a076d6f646c6163612f63706f740000000000000000000000000000000000000000659824ba0000000000000000000000000000020000000e0388dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eef9f9b6a203000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000e014f20900000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9ae10bf9cd0e372c142b2ca90219666906d6f646c6163612f747273790000000000000000000000000000000000000000",
    "0x00000000000000000e00000000000000cf43076f0cfa4e000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7df1daeb8986837f21cc5d17596bb78d19eb2dcce60f37a2702000000",
    "0x49024102840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee0009800bf65f366e8885494845a4fc56bba3a89172b8e2592b580ecb3f11d4e534ff469dcabdb1058805ea6035a7610ac32d6a1f8a92f5579b5f4fb69e3ad5c30b040100000a0000d17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae69070010a5d4e8",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef734abf5cb34d6244378cddbf18e849d96",
    "0x70e2160f000000000000000000000000189b6fa302000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a",
    "0x0003000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7a86da5a932684f199539836fcb8c886f",
    "0x89060000",
  ],
]
`;

exports[`dry-run-extrinsic > dry run extrinsic with fake signature 1`] = `
{
  "Ok": {
    "Ok": [],
  },
}
`;

exports[`dry-run-extrinsic > dry run extrinsic with fake signature 2`] = `
[
  [
    "0x26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850",
    "0x0b000000",
  ],
  [
    "0x3a7472616e73616374696f6e5f6c6576656c3a",
    undefined,
  ],
  [
    "0x3a65787472696e7369635f696e646578",
    "0x03000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9280b197190fc1a763257f65171ea366e88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
    "0x01000000000000000100000000000000005039278c0400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7ff553b5a9862a516939d82b3d3d8661a",
    "0x0003000000",
  ],
  [
    "0x1da53b775b270400e7e61ed5cbc5a146c726478796bad0b9cabd7481dbe64983",
    undefined,
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef780d41e5e16056765bc8461851072c9d7",
    "0x2c020b0a00016d6f646c6163612f63647074000000000000000000000000000000000000000016cd1b0000000000000000000000000000020b0000016d6f646c6163612f63647074000000000000000000000000000000000000000016cd1b0000000000000000000000000000020b0a00016d6f646c6163612f636470740000000000000000000000000000000000000000e737fa0100000000000000000000000000020b0a00016d6f646c6163612f6364707400000000000000000000000000000000000000004d125c100000000000000000000000000000000000000000001c6c090000000002000000010000000000000000000000000002000000020000000a0888dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eee3669df1020000000000000000000000000002000000ff00000000020000000a0788dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eee3669df10200000000000000000000000000020000000e0388dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000001019062a01000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7df1daeb8986837f21cc5d17596bb78d19eb2dcce60f37a2702000000",
    "0xc101b901840088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee01deadbeefcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd000000ff00000080969800",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef734abf5cb34d6244378cddbf18e849d96",
    "0xa0e62a2f010000000000000000000000189b6fa302000000",
  ],
  [
    "0xc2261276cc9d1f8598ea4b6a74b15c2f57c875e4cff74148e4628f264b974c80",
    "0x589c8e73199464c80000000000000000",
  ],
  [
    "0x26aa394eea5630e07c48ae0c9558cef7a86da5a932684f199539836fcb8c886f",
    "0x67060000",
  ],
]
`;
